src/Api/ExtendedFormatsProviderInterface.php

```diff
@@ -0,0 +1,27 @@
+<?php
+
+/*
+ * This file is part of the API Platform project.
+ *
+ * (c) Kévin Dunglas <dunglas@gmail.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace ApiPlatform\Core\Api;
+
+/**
+ * Extracts formats for a given operation according to the retrieved Metadata.
+ *
+ * @author Anthony GRASSIOT <antograssiot@free.fr>
+ */
+interface ExtendedFormatsProviderInterface
```

Comment:
I'm clear not a fan either on the name either on the idea.
the FormatProviderInterface as added in latest version IIRC, I did it the "clean" way regarding BC I think but I would be in inclined to update the existing interface, it is very quite unlikely that someone already implemented it... WDYT ?

Reply:
At least it could extend `FormatsProviderInterface`. For the name, may I suggest `OperationAwareFormatsProviderInterface`?

Reply:
indeed, updated.